src/Hl7.Fhir.Core/ElementModel/PocoBuilderExtensions.cs

```diff
@@ -16,14 +16,17 @@ namespace Hl7.Fhir.ElementModel
 {
     public static class PocoBuilderExtensions
     {
-        public static Base ToPoco(this ISourceNode source, Type pocoType = null, PocoBuilderSettings settings = null) =>
-            new PocoBuilder(ModelInspector.ForAssembly(pocoType.Assembly), settings).BuildFrom(source, pocoType);
+        public static Base ToPoco(this ISourceNode source, ModelInspector inspector, Type pocoType = null, PocoBuilderSettings settings = null) =>
```

Comment:
This I do not understand. Let's discuss tomorrow ;-)

Reply:
It goes wrong with `bundle.ToTypedElement()` . `Bundle`'s type information can still be found in the big common, but... `Bundle` can also contain other resources, such as `Patient`. At that moment (if we are in the middle of the big common), then type information of `Patient` cannot be retrieved. It's somewhere in the satellites that common doesn't know about. Hence this overload.

Reply:
Well, that's a good one. But then we have not solved it, since this is true for other things than `Bundle`, like `Parameters`, where `Parameters.value[x]` and `Parameters.resource` can contain instances from a different satellite than common. And how would the user know when to call this overload instead of the other?

Reply:
Solved in PR #2272